PowerPointLabs/PowerPointLabs/Views/ShapesLabCategoryInfoDialogBox.xaml.cs

```diff
@@ -1,108 +1,90 @@
-        {
-            UserOption = Option.Cancel;
-
-            Dispose();
-        }
-
-        private void CategoryNameBoxKeyDown(object sender, KeyEventArgs e)
-        {
-            if (e.KeyValue == (char)Keys.Return)
-            {
-                e.Handled = true;
-                OkButtonClick(sender, null);
-            }
-        }
-        # endregion
-
-        # region Helper Functions
-        private bool VerifyName(string name)
-        {
-            if (name.Length > 255)
-            {
-                MessageBox.Show(TextCollection.ErrorNameTooLong);
-                return false;
-            }
-
-            var invalidChars = new Regex(InvalidCharsRegex);
-            
-            if (string.IsNullOrWhiteSpace(name) ||
-                invalidChars.IsMatch(name))
-            {
-                MessageBox.Show(TextCollection.ErrorInvalidCharacter);
-                return false;
-            }
-
-            return true;
-        }
-
-        private bool VerifyCategory(string name)
-        {
-            if (Globals.ThisAddIn.ShapePresentation.HasCategory(name))
-            {
-                MessageBox.Show(TextCollection.CustomShapeDuplicateCategoryNameError);
-                return false;
-            }
-
-            return true;
-        }
-        # endregion
-    }
-}
+﻿using System.Text.RegularExpressions;
+using System.Windows;
+
+using PowerPointLabs.ActionFramework.Common.Extension;
+
+namespace PowerPointLabs.Views
+{
+    /// <summary>
+    /// Interaction logic for ShapesLabCategoryInfoDialogBox.xaml
+    /// </summary>
+    public partial class ShapesLabCategoryInfoDialogBox
+    {
+        private const string InvalidCharsRegex = "[<>:\"/\\\\|?*]";
+
+        public enum Option
+        {
+            Ok,
+            Cancel
+        }
+
+        public Option UserOption { get; private set; }
+        public string CategoryName { get; private set; }
+        
+
+        public ShapesLabCategoryInfoDialogBox(string name)
+        {
+            InitializeComponent();
+            
+            if (!string.IsNullOrEmpty(name))
+            {
+                nameInput.Text = name;
+                nameInput.SelectAll();
+            }
+
+            UserOption = Option.Cancel;
+        }
+
+        private void OkButton_Click(object sender, RoutedEventArgs e)
+        {
+            string categoryName = nameInput.Text;
+
+            if (VerifyName(categoryName) &&
+                VerifyCategory(categoryName))
+            {
+                CategoryName = nameInput.Text;
+                UserOption = Option.Ok;
+                Close();
+            }
+        }
+
+        private void CancelButton_Click(object sender, RoutedEventArgs e)
+        {
+            UserOption = Option.Cancel;
+            Close();
+        }
+
+        #region Helper Functions
+        private bool VerifyName(string name)
+        {
+            if (name.Length > 255)
```

Comment:
Where does this 255 come from? Is it specific to this dialog? Or do other dialogs have the same length restriction?